lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -92,23 +94,25 @@ def preprocess(self, client_json: JSON) -> CONTEXT:
         object and run_id as JSON CONTEXT so that the postprocess operations
         can use it to identify the index to be searched from document index
         metadata.
+
+        Raises:
+        APIAbort: input can't be validated or normalized
         """
         run_id = client_json["run_id"]
 
         # Query the dataset using the given run id
         try:
             dataset = Dataset.query(md5=run_id)
         except DatasetNotFound:
-            self.logger.debug(f"Dataset with Run ID {run_id!r} not found")
-            abort(HTTPStatus.NOT_FOUND, message="Dataset not found")
-
+            raise APIAbort(
```

Comment:
FYI, this is one of the places where we don't have any `logger` message associated with *raising* `APIAbort`. I'd rather not depend on this; it's nice to have that handled in common at the `except APIAbort`. The only problem with that is the `INTERNAL ERROR` messages, which don't add a lot.

Reply:
I suppose that we could either special case `HTTPStatus.INTERNAL_SERVER_ERROR` or just generalize the exception to include separate `debug_message` and `user_message` properties, with `debug_message` defaulting to `user_message` if only one is specified. (And `user_message` defaulting to `"INTERNAL ERROR"` for `HTTPStatus.INTERNAL_SERVER_ERROR`.) The main question being whether we really would ever want separate messages *except* in that case. 👀